### ISSUE TITLE
Inconsistent get_lexer_for_filename() with XML

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#988]
(Original issue 1284 created by roskakori on 2016-09-15T21:07:51.223800+00:00)

For the same plain XML files, `pygments.lexers.get_lexer_for_filename` sometimes returns XML and other times XML+Evoque.

To reproduce this, first create a simple XML file:

```sh
$ echo >/tmp/some.xml '<?xml?><some/>'
```
Then make `pygments` print the appripriate lexer for it:
```sh
python3 -c "from pygments.lexers import get_lexer_for_filename; print(get_lexer_for_filename('/tmp/some.xml'));"
```
This sometimes output

> <pygments.lexers.XmlLexer>

and other times

> <pygments.lexers.EvoqueXmlLexer>

This reason probably is that `EvoqueXmlLexer` does not have a proper `analyse_text()` and both lexers assign the same value to it, so the one showing up earlier depending on Python's current hash seed wins.

I tried looking into adding an `EvoqueXmlLexer.analyse_text()` but sadly the web site http://evoque.gizmojo.org/ seems to timeout and the repository has not been updated for almost 4 years, so I was unable to find details about eh Evoque syntax.

**Comment by Anteru** [comment]
(Original issue 1284 last updated on 2017-01-22T17:42:29.220932+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by roskakori on 2016-10-01T05:12:29.449433+00:00)

I added pull request #669 which provides a more general fix that would also solve this issue.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2017-01-22T17:42:29.208933+00:00)

Changed find_lexer_class_for_filename() to use the lexer name as secondary criteria when sorting for rating. This ensures that even when multiple lexers return the same rating the same lexer wins. Before the lexer with the lowest hashcode in the LEXERS dictionary won, which could return different results between repeated invocations of Python. This also fixes issue #1284.

→ <<cset b9c6a8d83fc9>>
